apps/backend/src/modules/projects/project.controller.ts

```diff
@@ -210,12 +216,18 @@ class ProjectController extends BaseController {
 	): Promise<APIHandlerResponse> {
 		const { query } = options;
 
+		if (!query.name) {
+			return {
+				payload: await this.projectService.findAll(),
+				status: HTTPCode.OK,
+			};
+		}
```

Comment:
instead let's merge findAll and findByName methods for projects service and projects repository and keep only findAll. For repository type let's define find like the following. And inside projects repository we need to do the same as we did for findByEmail method in users repository - in case the query is true, only then we attach `whereILike` to the query
```suggestion
  findAll(payload: unknown): Promise<{ items: T[] }>;
```

Reply:
Just to make sure, before i do any more changes, do you mean findByName, or the findAllbyName?

Reply:
I mean findAllbyName